Kernel/Signature.cpp

```diff
@@ -697,6 +697,43 @@ unsigned Signature::getDiff(){
 }
 
 
+unsigned Signature::getFnDef(unsigned fn)
+{
+  auto type = getFunction(fn)->fnType();
+  auto sort = type->result();
+  bool added = false;
+  auto name = "$def_"+sort.toString();
```

Comment:
Thats's a good question. I thought it would be better to use a single definition for functions that have the same result, but maybe it's easier (and more robust) to have a definition for each.